### PR TITLE
Add VCS repo filtering

### PR DESCRIPTION
**Pull request by alextu**
This PR adds a configuration field to specify patterns against which to check the VCS repo URL.
This can either be include or exclude patterns.

image

### CODE REVIEW

File: src/main/java/com/gradle/develocity/bamboo/DevelocityPreBuildAction.java [hunk 1]
```diff
@@ -38,25 +46,46 @@ public ErrorCollection validate(BuildConfiguration configuration) {
...
+                }
             }
         }
 
         return buildContext;
     }
 
+    private boolean injectionIsAllowedOnVcsRepo() {
+        // TODO: we don't need GradleConfiguration
+        //  need to remove inheritance and use composition for BuildToolConfiguration, and instantiate that here
```
Comment (welandaz) [inline src/main/java/com/gradle/develocity/bamboo/DevelocityPreBuildAction.java:66]: Do you plan to resolve it in this PR?

File: src/main/java/com/gradle/develocity/bamboo/Utils.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+class Utils {
+
+    private Utils() {
+    }
+
+    static Set<String> vcsRepoUrls(BuildContext buildContext) {
```
Comment (welandaz) [inline src/main/java/com/gradle/develocity/bamboo/Utils.java:14]: I'd make this method part of `DevelocityPreBuildAction` class. Introducing this `Utils` class seems very artificial and it's better to bind this logic to some domain entity. 

If extracting, I'd extract the whole vcs decision logic separately and use it as a component in `DevelocityPreBuildAction`; So either all there via methods or all out in it's own entity.

**Review by welandaz — APPROVED** [review]
All good, left some comments

File: src/main/java/com/gradle/develocity/bamboo/DevelocityPreBuildAction.java (hunk 1 above)
Comment (alextu) [reply src/main/java/com/gradle/develocity/bamboo/DevelocityPreBuildAction.java:66]: ah no, will add a comment, we'll refactor it if we encounter the same again

**Pull request merged**